api/tests/unit/features/feature_segments/test_unit_feature_segments_views.py

```diff
@@ -28,29 +30,37 @@
 
 
 @pytest.mark.parametrize(
-    "client, num_queries",
+    "client,is_admin_master_api_key_client, num_queries",
     [
         (
             lazy_fixture("admin_client"),
+            False,
             6,
         ),  # 1 for paging, 3 for permissions, 1 for result, 1 for getting the current live version
         (
             lazy_fixture("admin_master_api_key_client"),
+            True,
             4,
         ),  # an extra one for master_api_key
     ],
 )
 def test_list_feature_segments(
-    segment,
-    feature,
-    environment,
-    project,
-    django_assert_num_queries,
-    client,
-    feature_segment,
-    num_queries,
+    segment: Segment,
+    feature: Feature,
+    environment: Environment,
+    project: Project,
+    django_assert_num_queries: DjangoAssertNumQueries,
+    client: APIClient,
+    feature_segment: FeatureSegment,
+    num_queries: int,
+    is_admin_master_api_key_client: bool,
 ):
     # Given
+    if (
+        settings.IS_RBAC_INSTALLED and not is_admin_master_api_key_client
```

Comment:
This feels slightly odd to me: 

 1. I'd rather not have to use another parametrized argument for the boolean if we can avoid it (although I have looked into it myself and can't see any obvious options). 
 2. If we do need to use the extra argument, why not invert it and use `is_admin_client` rather than `is_admin_master_api_key_client`? 

Either way, it'd be good to add a comment as to why we expect the extra query in this scenario.